dash_auth/oidc_auth.py

```diff
@@ -253,29 +257,35 @@ def logout(self):  # pylint: disable=C0116
         return page
 
     def callback(self, idp: str):  # pylint: disable=C0116
-        """Do the OIDC dance."""
+        """Handle the OIDC dance and post-login actions."""
         if idp not in self.oauth._registry:
             return f"'{idp}' is not a valid registered idp", 400
 
         oauth_client = self.get_oauth_client(idp)
         oauth_kwargs = self.get_oauth_kwargs(idp)
         try:
-            token = oauth_client.authorize_access_token(
-                **oauth_kwargs.get("authorize_token_kwargs", {}),
+            self.token = oauth_client.authorize_access_token(
+                **oauth_kwargs.get("authorize_token_kwargs", {})
             )
         except OAuthError as err:
             return str(err), 401
-        user = token.get("userinfo")
+
+        user = self.token.get("userinfo")
+        return self.after_logged_in(user, idp)
+
+    def after_logged_in(self, user: dict | None, idp: str):
+        """Post-login actions after successful OIDC authentication."""
         if user:
             session["user"] = user
             session["idp"] = idp
-            if "offline_access" in oauth_client.client_kwargs["scope"]:
-                session["refresh_token"] = token.get("refresh_token")
+            if "offline_access" in self.oauth._registry[idp][1].get("client_kwargs").get("scope"):
```

Comment:
Should use `self.get_oauth_client(idp)` to get the oauth_client like in previous function